### PR TITLE
PIP-383 Spring Boot 3 and spring 6 upgrade 

### PR DESCRIPTION
**Pull request by QuyenLy87**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -84,6 +97,29 @@
...
+			<artifactId>jaxb-impl</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.javassist</groupId>
+			<artifactId>javassist</artifactId>
+			<version>3.25.0-GA</version>
+		</dependency>
+		<dependency>
+			<artifactId>netty-nio-client</artifactId>
+			<groupId>software.amazon.awssdk</groupId>
```
Comment (CoderMChu) [inline pom.xml:120]: @QuyenLy87 Do we need these dependencies?

File: src/main/java/org/ihtsdo/buildcloud/config/Config.java [hunk 2]
```diff
@@ -106,11 +103,9 @@ public HibernateTransactionManager transactionManager(
 	}
 
 	@Bean
-	public S3ClientFactory s3ClientFactory(@Value("${aws.key}") final String accessKey,
-	                                       @Value("${aws.privateKey}") final String privateKey,
-	                                       @Value("${srs.build.s3.offline.directory}") final String directory) throws IOException {
+	public S3ClientFactory s3ClientFactory(@Value("${srs.build.s3.offline.directory}") final String directory) throws IOException {
 		s3ClientFactory = new S3ClientFactory();
-		s3ClientFactory.setOnlineImplementation(new S3ClientImpl(new BasicAWSCredentials(accessKey, privateKey)));
+		s3ClientFactory.setOnlineImplementation(new S3ClientImpl(software.amazon.awssdk.services.s3.S3Client.builder().credentialsProvider(ProfileCredentialsProvider.create()).build()));
```
Comment (CoderMChu) [inline src/main/java/org/ihtsdo/buildcloud/config/Config.java:108]: @QuyenLy87 Can we use default region builder instead please?

File: src/main/java/org/ihtsdo/buildcloud/config/SecurityContextConfiguration.java [hunk 3]
```diff
@@ -5,26 +5,32 @@
...
 	@Bean
 	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-		http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);
-		http.authorizeRequests()
-				.antMatchers("/snomed-release-service-websocket/**/*",
+		http.httpBasic(withDefaults());
+		http.csrf(AbstractHttpConfigurer::disable);
+		http.sessionManagement(sessionManagement -> sessionManagement.sessionCreationPolicy(SessionCreationPolicy.STATELESS));
+		http.addFilterBefore(new RequestHeaderAuthenticationDecorator(), FilterSecurityInterceptor.class);
+		http.authorizeHttpRequests((authorize) -> authorize.requestMatchers(
```
Comment (CoderMChu) [inline src/main/java/org/ihtsdo/buildcloud/config/SecurityContextConfiguration.java:25]: During authoring services testing /**/* won't work. You either nee to use ant path matcher or use /**/ only

File: src/test/java/org/ihtsdo/buildcloud/TestConfig.java [hunk 4]
```diff
@@ -1,32 +1,127 @@
...
+	@Bean(name = "offlineS3Client")
+	@DependsOn("s3ClientFactory")
+	public S3Client s3Client(@Value("${srs.build.offlineMode}") final boolean offlineMode) {
+		return s3ClientFactory.getClient(offlineMode);
+	}
+
+	@Bean
+	public S3ClientFactory s3ClientFactory(@Value("${srs.build.s3.offline.directory}") final String directory) throws IOException {
+		s3ClientFactory = new S3ClientFactory();
+		s3ClientFactory.setOnlineImplementation(new S3ClientImpl(software.amazon.awssdk.services.s3.S3Client.builder().credentialsProvider(ProfileCredentialsProvider.create()).build()));
```
Comment (CoderMChu) [inline src/test/java/org/ihtsdo/buildcloud/TestConfig.java:71]: We don't need to use online implementation for testing unless we are using local stack for S3

**Review by CoderMChu — COMMENTED** [review]
@QuyenLy87 It looks good apart from minor changes. Thanks

File: src/main/java/org/ihtsdo/buildcloud/config/Config.java (hunk 2 above)
Comment (QuyenLy87) [reply src/main/java/org/ihtsdo/buildcloud/config/Config.java]: The code has been updated. Thanks @CoderMChu

File: src/main/java/org/ihtsdo/buildcloud/config/SecurityContextConfiguration.java (hunk 3 above)
Comment (QuyenLy87) [reply src/main/java/org/ihtsdo/buildcloud/config/SecurityContextConfiguration.java:25]: The changes have been made. Thanks @CoderMChu for your help.

File: src/test/java/org/ihtsdo/buildcloud/TestConfig.java (hunk 4 above)
Comment (QuyenLy87) [reply src/test/java/org/ihtsdo/buildcloud/TestConfig.java]: This Bean configuration has been removed from TestConfig.java as redundant. @CoderMChu

**Pull request merged**